state/binarystorage.go

```diff
@@ -18,18 +18,16 @@ var binarystorageNew = binarystorage.New
 //
 // TODO(axw) remove this, add a constructor function in binarystorage.
 func (st *State) ToolsStorage() (binarystorage.StorageCloser, error) {
-	return newStorage(st, toolsmetadataC), nil
+	return newStorage(st.session.Copy(), st.ModelUUID(), toolsmetadataC), nil
```

Comment:
... but I think I'd probably just pass in the state as before and just add the uuid parameter, leaving the newStorage function responsible for cloning (or copying) the underlying session from within the state.